signer/tests/integration/transaction_coordinator.rs

```diff
@@ -673,166 +493,315 @@ async fn process_complete_deposit() {
+                    Ok(shares.map(|sh| sh.aggregate_key))
+                })
+            });
+        // The coordinator broadcasts a rotate keys transaction if it
+        // is not up-to-date with their view of the current aggregate
+        // key. The response of None means that the stacks node does
+        // not have a record of a rotate keys contract call being
+        // executed, so the coordinator will construct and broadcast
+        // one.
+        client
+            .expect_get_current_signers_aggregate_key()
+            .returning(move |_| Box::pin(std::future::ready(Ok(None))));
+
+        // Only the client that corresponds to the coordinator will
+        // submit a transaction, so we don't make explicit the
+        // expectation here.
+        client.expect_submit_tx().returning(move |tx| {
+            let tx = tx.clone();
+            let txid = tx.txid();
+            let broadcast_stacks_tx = broadcast_stacks_tx.clone();
+            Box::pin(async move {
+                broadcast_stacks_tx.send(tx).unwrap();
+                Ok(SubmitTxResponse::Acceptance(txid))
+            })
+        });
+        // The coordinator will get the total supply of sBTC to
+        // determine the amount of mintable sBTC.
+        client
+            .expect_get_sbtc_total_supply()
+            .returning(move |_| Box::pin(async move { Ok(Amount::ZERO) }));
+
+        client
+            .expect_is_deposit_completed()
+            .returning(move |_, _| Box::pin(async move { Ok(false) }));
+
+        // We use this during validation to check if the withdrawal
+        // request completed in the smart contract.
+        client
+            .expect_is_withdrawal_completed()
+            .returning(|_, _| Box::pin(std::future::ready(Ok(false))));
+    })
+    .await;
+}
+
+/// Tests that the coordinator deploys the smart contracts in the correct
+/// order if none are deployed.
 #[tokio::test]
-async fn deploy_smart_contracts_coordinator<F>(
-    smart_contracts: &[SmartContract],
-    stacks_client_mock: F,
-) where
-    F: FnOnce(u64, Sender<StacksTransaction>) -> Box<dyn FnOnce(&mut MockStacksInteract)>,
-{
-    let db = testing::storage::new_test_database().await;
-    let mut rng = get_rng();
+async fn deploy_smart_contracts_coordinator() {
+    let (_, signer_key_pairs): (_, [Keypair; 3]) = testing::wallet::regtest_bootstrap_wallet();
+    let (rpc, faucet) = regtest::initialize_blockchain();
 
-    let num_messages = smart_contracts.len();
+    signer::logging::setup_logging("info,signer=debug", false);
 
-    let bitcoin_block: model::BitcoinBlock = Faker.fake_with_rng(&mut rng);
-    db.write_bitcoin_block(&bitcoin_block).await.unwrap();
+    // We need to populate our databases, so let's fetch the data.
+    let emily_client = EmilyClient::try_new(
+        &Url::parse("http://testApiKey@localhost:3031").unwrap(),
+        Duration::from_secs(1),
+        None,
+    )
+    .unwrap();
```

Comment:
nit: do we need the real Emily for this test?

Reply:
Kinda, this test spins up a block observer so we need either a real emily or a mocked one and a real one is easier.